src/framing/FrameTransport.cpp

```diff
@@ -69,10 +69,23 @@ void FrameTransport::setFrameProcessor(
   }
 }
 
-void FrameTransport::close(folly::exception_wrapper ex) {
+void FrameTransport::close() {
+  closeImpl<false>(folly::exception_wrapper());
+}
+
+void FrameTransport::closeErr(folly::exception_wrapper ew) {
+  if (!ew) {
+    VLOG(1) << "FrameTransport::closeErr() called with empty exception";
+    ew = std::runtime_error("Undefined error");
+  }
+  closeImpl<true>(std::move(ew));
+}
+
+template <bool isErr>
```

Comment:
I think this could use a bool arg instead, I don't see the reason for this to use a template. Or just use `folly::Optional`.